src/main/java/nextstep/subway/section/Sections.java

```diff
@@ -17,19 +16,195 @@ public class Sections {
 
     public Sections() { }
 
+    public List<Map<String, Object>> getStationsResponse() {
+        return getStations().stream()
+                .map(Station::toMapForOpen)
+                .collect(Collectors.toList());
+    }
+
+    private Set<Station> getStations() {
+        sort();
+        Set<Station> stations = new LinkedHashSet<>();
+        for (Section section : sections) {
+            stations.add(section.getUpStation());
+            stations.add(section.getDownStation());
+        }
+        return stations;
+    }
+
+    private void sort() {
+        sections.sort(Comparator.comparingInt(Section::getSequence));
+    }
+
     public void add(Section section) {
-        sections.add(section);
+        if (addFirstSection(section)) { return; }
+        validateStationName(section);
+        sort();
+        if (extendDownSection(section)) { return; }
+        if (extendUpSection(section)) { return; }
+        if (sliceDownSection(section)) { return; }
+        if (sliceUpSection(section)) { return; }
+        throw new IllegalArgumentException("구간 등록이 불가능한 알 수 없는 오류입니다.");
     }
 
-    public List<Map<String, Object>> getStations() {
+    private boolean addFirstSection(Section section) {
         if (sections.size() == 0) {
```

Comment:
```suggestion
        if (sections.isEmpty()) {
```

isEmpty 가 조금 더 의미 전달에 좋지 않을까요?